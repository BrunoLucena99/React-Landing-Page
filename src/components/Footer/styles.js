import styled from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';
import { Container as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.footer`
  text-align: center;
  border-top: ${({ theme }) => `0.1rem solid ${theme.colors.mediumGray}`};

  a {
    color: inherit;
    text-decoration: none;
  }

  ${TextComponent} {
    font-size: ${({ theme }) => theme.font.sizes.small};
  }

  ${SectionContainer} {
    padding-top: 0;
    padding-bottom: 0;
  }
`;
