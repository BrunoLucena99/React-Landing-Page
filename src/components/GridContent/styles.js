import styled from 'styled-components';

export const Container = styled.div`
  text-align: center;
  max-width: 58rem;
  margin: 0 auto;
`;

export const HTMLContainer = styled.div`
  margin: ${({ theme }) => theme.spacings.xhuge} 0;
`;
