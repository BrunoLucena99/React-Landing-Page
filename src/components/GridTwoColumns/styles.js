import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacings.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    gap: ${theme.spacings.xlarge};

    @media ${theme.media.lteMedium} {
      margin-bottom: ${theme.spacings.large};
      text-align: center;
    }
  `}
`;

export const ImageContainer = styled.div``;

export const Image = styled.img``;
