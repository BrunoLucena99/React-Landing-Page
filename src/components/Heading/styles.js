import styled, { css } from 'styled-components';

const titleSizes = {
  small: ({ font }) => css`
    font-size: ${font.sizes.medium};
  `,
  medium: ({ font }) => css`
    font-size: ${font.sizes.large};
  `,
  big: ({ font }) => css`
    font-size: ${font.sizes.xlarge};
  `,
  huge: (theme) => css`
    font-size: ${theme.font.sizes.xhuge};
    ${mediaFont(theme)}
  `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.xlarge};
  }
`;

export const Title = styled.h1`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSizes[size](theme)};
    text-transform: ${uppercase ? 'uppercase' : 'none'};
  `}
`;
