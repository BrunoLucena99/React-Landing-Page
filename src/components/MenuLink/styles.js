import styled from 'styled-components';

export const Container = styled.a`
  font-size: ${({ theme }) => theme.font.sizes.small};
  display: block;
  text-decoration: none;
  padding: ${({ theme }) => theme.spacings.small};
  color: ${({ theme }) => theme.colors.primaryColor};
  position: relative;

  &::after {
    content: '';
    position: absolute;
    bottom: 0;
    left: 50%;
    width: 0;
    height: 0.2rem;
    background: ${({ theme }) => theme.colors.secondaryColor};
    transition: all 300ms ease-in-out;
  }

  &:hover::after {
    left: 25%;
    width: 50%;
  }
`;
