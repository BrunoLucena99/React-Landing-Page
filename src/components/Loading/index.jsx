import { Container } from './styles';

const Loading = () => <Container />;

export default Loading;
