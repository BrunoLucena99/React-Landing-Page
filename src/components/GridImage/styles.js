import styled from 'styled-components';
import { Title as HeadingTitle } from '../Heading/styles';

export const Container = styled.div``;

export const Grid = styled.div`
  display: grid;
  gap: ${({ theme }) => theme.spacings.large};
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  margin-top: ${({ theme }) => theme.spacings.xhuge};
`;

export const GridElement = styled.div`
  overflow: hidden;
`;

export const Image = styled.img`
  width: 100%;
  transition: all 300ms ease-in-out;

  &:hover {
    transform: scale(1.2) rotate(10deg);
  }
`;
