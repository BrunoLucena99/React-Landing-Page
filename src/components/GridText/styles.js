import styled from 'styled-components';
import { Title as HeadingTitle } from '../Heading/styles';

export const Container = styled.div``;

export const Grid = styled.div`
  counter-reset: grid-counter;
  display: grid;
  gap: ${({ theme }) => theme.spacings.large};
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  margin-top: ${({ theme }) => theme.spacings.xhuge};
`;

export const GridElement = styled.div`
  ${HeadingTitle} {
    position: relative;
    left: 5rem;
  }

  ${HeadingTitle}::before {
    counter-increment: grid-counter;
    content: counter(grid-counter);
    position: absolute;
    font-size: 7rem;
    top: -3rem;
    left: -5rem;
    transform: rotate(5deg);
  }
`;
