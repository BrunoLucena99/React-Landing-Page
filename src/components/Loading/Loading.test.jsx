import { renderTheme } from '../../styles/RenderTheme';
import Loading from '.';
import { screen } from '@testing-library/react';
import { theme } from '../../styles/theme';

describe('<Loading />', () => {
  it('should render', () => {
    const { container } = renderTheme(<Loading />);
    expect(container).toMatchSnapshot();
  });
});
