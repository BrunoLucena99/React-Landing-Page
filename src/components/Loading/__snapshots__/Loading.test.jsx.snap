// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should render 1`] = `
.c0 {
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  z-index: 10;
  background: #0A1128;
}

.c0:before,
.c0:after {
  content: '';
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  border: 0.5rem solid transparent;
  border-radius: 50%;
}

.c0:after {
  width: 6rem;
  height: 6rem;
  border-left: 0.5rem solid #dc143c;
  border-top: 0.5rem solid #dc143c;
  -webkit-animation: jSikFv 600ms linear infinite;
  animation: jSikFv 600ms linear infinite;
}

.c0:before {
  width: 2rem;
  height: 2rem;
  border-left: 0.5rem solid #dc143c;
  border-top: 0.5rem solid #dc143c;
  -webkit-animation: jSikFv 1s linear reverse infinite;
  animation: jSikFv 1s linear reverse infinite;
}

<div>
  <div
    class="c0"
  />
</div>
`;
